src/tests/ffi.cpp

```diff
@@ -5923,3 +5925,36 @@ TEST_F(rnp_tests, test_ffi_security_profile)
 
     rnp_ffi_destroy(ffi);
 }
+
+TEST_F(rnp_tests, test_result_to_string)
+{
+    const char *          result_string = NULL;
+    rnp_result_t          code;
+    std::set<std::string> stringset;
+
+    result_string = rnp_result_to_string(RNP_SUCCESS);
+    assert_string_equal(result_string, "Success");
+
+    /* Cover all defined error code ranges,
+     * check that each defined
+     * code has corresponding unique string */
+
+    std::vector<std::pair<rnp_result_t, rnp_result_t>> error_codes = {
+      {RNP_ERROR_GENERIC, RNP_ERROR_NULL_POINTER},
+      {RNP_ERROR_ACCESS, RNP_ERROR_WRITE},
+      {RNP_ERROR_BAD_STATE, RNP_ERROR_SIGNATURE_UNKNOWN},
+      {RNP_ERROR_NOT_ENOUGH_DATA, RNP_ERROR_EOF}};
+
+    for (auto range = error_codes.begin(); range != error_codes.end(); ++range) {
+        for (code = range->first; code <= range->second; code++) {
+            result_string = rnp_result_to_string(code);
+
+            auto search = stringset.find(result_string);
+
+            /* Make sure returned error string is not already returned for other codes */
+            assert_true(search == stringset.end());
```

Comment:
We should also check that the string is not `Unknown error`.
And not NULL while we're at that.

Reply:
+1 for `Unknown error.`. Check for NULL is done automatically by not crashing on conversion from `char *` to `std::string`.

Reply:
"Unknown error" is returned in two cases, as a string result corresponding to RNP_ERROR_GENERIC parameter and as a default value for unknown codes.

Reply:
So, "Unknown error" is one of the possible values. Probably we can return different strings for default unknown code and RNP_ERROR_GENERIC, but it can break compatibility with existing applications.

Reply:
@ni4 what do you think about "Unknown error" comments above?

Reply:
@antonsviridenko we may change message for the unknown error code to something like "Unsupported error code". Existing implementations should not rely on error codes which we didn't return at all.